app/models/user.rb

```diff
@@ -1,8 +1,156 @@
 # frozen_string_literal: true
+
+require "net/ldap/dn"
+require "velum/ldap"
+
 # User represents administrators in this application.
+# rubocop:disable ClassLength
 class User < ApplicationRecord
-  enabled_devise_modules = [:database_authenticatable, :registerable,
+  enabled_devise_modules = [:ldap_authenticatable, :registerable,
                             :rememberable, :trackable, :validatable].freeze
 
   devise(*enabled_devise_modules)
+
+  before_create :create_ldap_user
+
+  protected
+
+  # rubocop:disable AbcSize,CyclomaticComplexity,MethodLength,PerceivedComplexity
+  def create_ldap_user
```

Comment:
I considered that, but, I didn't feel it would help readability and maintainability of the code to break it up. You'd end up with this function just calling the others, and the others are not intended to be reusable. It's a trade-off between being able to read the code sequentially (as this really is sequential code) or jumping around to do that. Alex Barclay and I talked about that before I took the WIP off this PR because I needed to bounce that off someone else.